File: cakk-api/src/main/java/com/cakk/api/dto/request/shop/CakeShopSearchRequest.java
package com.cakk.api.dto.request.shop;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;

public record CakeShopSearchRequest(
	Long cakeShopId,
	String keyword,
	@Min(-90) @Max(90)
	Double latitude,
	@Min(-180) @Max(180)
	Double longitude,
	Integer pageSize
) {
}

File: cakk-api/src/main/java/com/cakk/api/dto/request/shop/CakeShopSearchRequest.kt
package com.cakk.api.dto.request.shop

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min

import com.cakk.common.utils.DEFAULT_PAGE_SIZE

data class CakeShopSearchRequest(
    val cakeShopId: Long?,
    val keyword: String?,
	@field:Min(-90) @field:Max(90)
    val latitude: Double?,
	@field:Min(-180) @field:Max(180)
    val longitude: Double?,
    val pageSize: Int? = DEFAULT_PAGE_SIZE
)